src/v/kafka/protocol/schemata/generator.py

```diff
@@ -805,60 +862,82 @@ class response;
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 
-{% macro version_guard(field) %}
-{%- set cond = field.versions().guard() %}
-{%- if cond %}
+{% macro version_guard(field, flex) %}
+{%- set guard_enum = field.versions().guard_enum %}
+{%- set e, cond = field.versions().guard(flex, first_flex) %}
+{%- if e == guard_enum.NO_GUARD %}
+{{- caller() }}
+{%- elif e == guard_enum.NO_SOURCE %}
+{%- elif e == guard_enum.GUARD %}
```

Comment:
everything looks pretty good in the generator. eliminating some conditionals seems like a nice optimization but when the predominant apis like produce/consume are moving gigabytes it's ultimately lost in the noise. the main concern is: how do we gain confidence in the correctness of the optimization? i guess we could manually inspect a few generated requests.

Reply:
Yes I am in the process of doing a full manual audit

Reply:
I finished this earlier today, I didn't notice anything suspect

Reply:
Just finished, 40 files have to be audited all looks good. You can run this command `grep flexibleVersions src/v/kafka/protocol/schemata/*.json | grep -v none` to see how many schemas will be affected. Any schema file with `flexibleVersions: none` should produce code identical as it once was.